File: C_Programming/Assignment/Assignment5/strongfun.c
#include<stdio.h>
void isStrong(int num);
void main()
{
  int num;
  printf("Enter the no:");
  scanf("%d",&num);
  isStrong(num);
}
void isStrong(int num)
{
  int rem,fact,sum=0,i;
   int temp=num;
  while(num>0)
  {
    rem = num % 10;
    fact=1,i=1;
    while(i<=rem)
    {
    fact=fact*i;
    i++;
    }
    sum=sum+fact;
    num=num/10;
    
  }
  if(temp == sum)
  printf("It is a Strong Number",num);
  else
  printf("It is not Strong number",num);
}

//output:Enter the no:145
//It is a Strong Number
// Enter the no:134
// It is not Strong number

File: Java_Programs/Assignment5Type(4)/Strongfun.java
package Assignment5Type4;

public class Strongfun {

	public static void main(String[] args) {
		int no=145;
//	    printf("Enter the no:");
//	    scanf("%d",&no);
	    int res=isStrong(no);
	    if(res==1)
	    {
	        System.out.printf("the number is strong");
	    }
	    else
	    {
	        System.out.printf("the number is not strong");
	    }

	}
	
	public static int isStrong(int n)
	{
	    int temp=n,rem,sum=0;
	    while(n>0)
	    {
	        rem=n%10;
	        sum=sum + fact(rem);
	        n=n/10;
	        // printf("%d",sum);
	        
	    }
	    if(temp==sum)
	    	return 1;
	    else
	    	return 0;
	    //return (temp==sum);
	}
	public static int fact(int no)
	{
	    int fact=1;
	    for(int i=1;i<=no;i++)
	    fact=fact*i;
	    return fact;
	}
}
